### ISSUE TITLE
How to actually input lm and rm to this dspy??

### ISSUE THREAD

**Issue by SYV969** [stanfordnlp/dspy#232]
Hi, I've tried to implement a simple model 'deepset/bert-base-cased-squad2' as lm and squadv2 as training dataset. 
But it wasn't successful. The error message is shown as :
![Screenshot 2023-11-24 at 4 31 38 PM](https://github.com/stanfordnlp/dspy/assets/114745328/ea301c06-452a-45b0-8558-75d0b9a3e231)

Is there any examples that shows how a user of DSPy can train an individual dataset using a different LM rather than GPT3.5, or is there any example to use other HF model?

Given the intro.ipynb is a cache result when I tried to run the model with gpt-3.5-turbo-instruct the accuracy is actually lower than meta-llama/Llama-2-13b-chat-hf
<img width="750" alt="Screenshot 2023-11-24 at 4 39 40 PM" src="https://github.com/stanfordnlp/dspy/assets/114745328/179e5a73-deb8-448a-ae75-fb34918dbb59">

I'm urged to understand how to train this DSPy with other llm or rm and training dataset, Thanks!

**Comment by okhat** [comment]
Hey, thanks for opening the issue.

What is your goal overall? We're happy to help more directly if we understand what you'd like to achieve.

To your specific issue, `deepset/bert-base-cased-squad2` is not a generative LM. It can't be used to generate text, so it won't work for DSPy.

**Comment by okhat** [comment]
For the screenshot at the bottom (the one with Best score: 16.0), what are you compiling here?

For a notebook on finetuning, see examples/qa/*

**Comment by SYV969** [comment]
Hi, the bottombest score is using model = 'gpt-3.5-turbo-instruct', my question to this is, if I'm trying to train my own dataset do I have to change the gpt3.py to remove the cache response??
